src/deepsparse/pipeline.py

```diff
@@ -440,6 +446,40 @@ def validate(self):
         elif isinstance(router_validation, str):
             raise ValueError(f"Invalid Router for operators: {router_validation}")
 
+    def run_func(
+        self,
+        *args,
+        operator: Operator,
+        func: Callable,
+        inp: Any = None,
+        **kwargs,
+    ):
+        """
+        Wrap the operator with middleware and execute the func callable.
+        InferenceState, PipelineState is inside kwargs
+
+        :param operator: Operator instance
+        :param func: Desired function to call. Ex. SchedulerGroup.submit
+        :param inp: Any input to the operator. Ex. IntSchema
+        """
+
+        # wrap the operator with the middleware, if any
+        wrapped_operator = operator
+        if self.middleware_manager is not None:
+            wrapped_operator = self.middleware_manager.wrap(operator)
+
+        kwargs["operator"] = wrapped_operator
+        if inp:
+            output = (
+                func(*args, **kwargs, **inp)
+                if isinstance(inp, dict)
+                else func(inp, *args, **kwargs)
+            )
```

Comment:
could we also check if `inp` is a list? and add them as positional args if they are? that will make this even more generic?